orth

```diff
@@ -181,17 +181,26 @@ analyze_in_docker() {
   # Setup temporary directories to mount into docker:
   temp_dir=$(mktemp -d)
 
-  netrc_file="$temp_dir/.netrc"
-  echo -e "machine atrifactory.example.com" >> $netrc_file
-  echo -e "login $artifactory_user_name" >> $netrc_file
-  echo -e "password $artifactory_password" >> $netrc_file
-  chmod 0600 $netrc_file
+  mount_netrc_option=""
+  if [ -n "$netrc_machine" ] && \
+     [ -n "$netrc_login" ] && \
+     [ -n "$netrc_password" ]
+  then
+    netrc_file="$temp_dir/.netrc"
+    touch $netrc_file
+    chmod 0600 $netrc_file
+    echo -e "machine $netrc_machine" >> $netrc_file
+    echo -e "login $netrc_login" >> $netrc_file
+    echo -e "password $netrc_password" >> $netrc_file
+
+    mount_netrc_option="-v $netrc_file:/root/.netrc"
+  fi
```

Comment:
For convenience maybe add an `else` block which echoes a message like "Not creating .netrc file as .netrc settings are incomplete.".